### PR TITLE
Set serializer in configure

### PR DESCRIPTION
**Pull request by hallgren**

### CODE REVIEW

File: lib/sandthorn/aggregate_root_base.rb [hunk 1]
```diff
@@ -108,10 +108,16 @@ def aggregate_find aggregate_id
...
           end
+          
+          if first_event_snapshot?(events)
+            transformed_snapshot_event = events.first.merge(event_args: Sandthorn.deserialize_snapshot(events.first[:event_data]))
+            events.shift
+          end
+
           transformed_events = events.map do |e|
-            e.merge(event_args: Sandthorn.deserialize(e[:event_data]))
+              e.merge(event_args: Sandthorn.deserialize(e[:event_data]))
```
Comment (jesjos) [inline lib/sandthorn/aggregate_root_base.rb]: Why add two extra spaces here?

File: lib/sandthorn.rb [hunk 2]
```diff
@@ -122,6 +110,44 @@ def event_stores
       def event_store=(store)
         @event_stores = EventStores.new(store)
       end
+
+      def serializer= block
```
Comment (jesjos) [inline lib/sandthorn.rb]: I really prefer parentheses around method args. Perhaps we should consider using Rubocop?

File: lib/sandthorn.rb [hunk 3]
```diff
@@ -122,6 +110,44 @@ def event_stores
...
+        @serializer = block if block.is_a? Proc
+      end
+
+      def deserializer= block
+        @deserializer = block if block.is_a? Proc
```
Comment (jesjos) [inline lib/sandthorn.rb]: Would be nice if it accepted blocks as well.

Comment (jesjos) [inline lib/sandthorn.rb]: Fughettit, I was reading sloppy, doesn't work with `=`-method

**Comment by jesjos** [comment]
Hey I thought sane default serializers was nicer than having control flow (return if this else that) in the serialization methods. Check out this commit https://github.com/Sandthorn/sandthorn/commit/46c5a78ff2b6dfdddbc2cf731705d4db20633228

**Comment by hallgren** [comment]
Cool, thanks!

**Pull request merged**